ide/che-core-ide-api/src/main/java/org/eclipse/che/ide/api/project/ProjectImpl.java

```diff
@@ -0,0 +1,158 @@
+/*******************************************************************************
+ * Copyright (c) 2012-2015 Codenvy, S.A.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Codenvy, S.A. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.che.ide.api.project;
+
+import org.eclipse.che.api.core.model.project.SourceStorage;
+import org.eclipse.che.api.core.model.project.fs.Folder;
+import org.eclipse.che.api.core.model.workspace.Project;
+import org.eclipse.che.api.core.model.workspace.ProjectProblem;
+import org.eclipse.che.api.core.rest.shared.dto.Link;
+import org.eclipse.che.api.workspace.shared.dto.ProjectConfigDto;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Implementation of Project
+ * @author gazarenkov
+ */
+public class ProjectImpl implements Project {
+
+    private Set<String> types;
+    private Map<String, List<String>> attributes;
+    private SourceStorage sourceStorage;
+    private String name;
+    private String path;
+    private String description;
+    private String type;
+    private List<String> mixins;
+    private List<ProjectImpl> modules;
+
+    ProjectImpl(ProjectConfigDto configDto, Set<String> types) {
+
+        this.types = types;
+        this.name = configDto.getName();
+        this.path = configDto.getPath();
+        this.description = configDto.getDescription();
+        this.type = configDto.getType();
+
+        if(configDto.getMixins() == null)
```

Comment:
*if*-blocks should be formatted in accordance with codestyle: https://github.com/codenvy/codenvy-dev-resources/wiki/Formatting#non-empty-blocks